ipython/fragment_reattachment_example.ipynb

```diff
@@ -0,0 +1,1102 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "d3349900",
+   "metadata": {},
+   "source": [
+    "# Random Reattachment of Fragment Species from Automatic Fragment Modeling (AFM) Simulation Results\n",
+    "AFM allows users to construct mechanisms for large molecules with RMG by breaking them up into two or more fragments. Running simulations with a fragment mechanism is much the same as any other mechanism. However, the species profiles obtained are for individual fragments that may be attached in many combinations. In this notebook, we give an example of random reattachment of fragments produced from polymer pyrolysis, which allows the user to obtain important observables such as molecular weight distribution or the approximate concentration of specific species composed of multiple fragments."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "285ac939-646d-46e6-a5b9-286661e1ad18",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:root:No normalization for BCUT2D_MWHI\n",
+      "WARNING:root:No normalization for BCUT2D_MWLOW\n",
+      "WARNING:root:No normalization for BCUT2D_CHGHI\n",
+      "WARNING:root:No normalization for BCUT2D_CHGLO\n",
+      "WARNING:root:No normalization for BCUT2D_LOGPHI\n",
+      "WARNING:root:No normalization for BCUT2D_LOGPLOW\n",
+      "WARNING:root:No normalization for BCUT2D_MRHI\n",
+      "WARNING:root:No normalization for BCUT2D_MRLOW\n",
+      "WARNING:root:No normalization for AvgIpc\n"
+     ]
+    }
+   ],
+   "source": [
+    "from rdkit.Chem.rdmolops import *\n",
+    "from rmgpy.molecule.fragment import Fragment\n",
+    "from rmgpy.tools.canteramodel import Cantera\n",
+    "from rmgpy.chemkin import *\n",
+    "import re\n",
+    "import os\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "import sys\n",
+    "rmgpy_loc = \"/home/gridsan/adoner/RMG-Py/\"\n",
+    "sys.path.append(rmgpy_loc)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "045d6ebe-c055-42bc-ae55-643599631ea3",
+   "metadata": {},
+   "source": [
+    "## Defining the class FragList\n",
+    "The `FragList` object stores molecule and fragment concentrations from a single time step and provides a series of functions that groups fragments to form molecules. The procedure is described later on in this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "9802cd48-4a7b-4244-acf8-cba0fc1aab69",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class FragList():\n",
+    "    '''\n",
+    "    to instantiate a FragList:\n",
+    "    fl = Fraglist(frag_list) \n",
+    "    where frag_list is a list of tuples of fragments and their amounts\n",
+    "    '''\n",
+    "\n",
+    "    def __init__(self, frag_list):\n",
+    "        self.raw = frag_list\n",
```

Comment:
`raw` could be named more informatively - maybe something like `unprocess_fragments`? `raw_fragment_output`? etc.